File: src/ch03/3.5.2_2_RegularExpressionsAndTriplequotedStrings1.ts
// Define a function to parse the path
function parsePath(path: string) {
    // Define a regular expression to match the path
    const regex = /(.+)\/(.+)\.(.+)/;
    // Match the entire path against the regex
    const matchResult = path.match(regex);
    // If there is a match, destructure the result
    if (matchResult) {
        const [_, directory, filename, extension] = matchResult;
        console.log(`Dir: ${directory}, name: ${filename}, ext: ${extension}`);
    }
}

// Main function to call parsePath with a sample path
function main() {
    parsePath("/Users/yole/kotlin-book/chapter.adoc");
}

// Call the main function
main();

File: src/ch03/3.5.2_2_RegularExpressionsAndTriplequotedStrings1.kt
// Define a type for the match result
type MatchResult = {
    directory: string;
    filename: string;
    extension: string;
};
// Function to parse the path
function parsePath(path: string): void {
    const regex = /(.+)\/(.+)\.(.+)/;
    const matchResult = regex.exec(path);
    if (matchResult !== null) {
        const [_, directory, filename, extension] = matchResult;
        console.log(`Dir: ${directory}, name: ${filename}, ext: ${extension}`);
    }
}
// Main function
function main(args: string[]): void {
    parsePath("/Users/yole/kotlin-book/chapter.adoc");
}